pkg/cli/cli.go

```diff
@@ -0,0 +1,100 @@
+package cli
+
+import (
+	"reflect"
+
+	"github.com/spf13/cobra"
+)
+
+type cobraCmd = *cobra.Command
+type cobraArgs = []string
+
+// Used for command flags
+type OkFlag struct {
+	Name      string
+	ShortName string // Optional
+	ValueType string
+	Usage     string
+}
+
+type OkCmd struct {
+	Use                    string
+	Short                  string
+	Long                   string
+	Run                    func(cmd cobraCmd, args cobraArgs)
+	Flags                  []OkFlag
+	Aliases                []string
+	EnforceFlagConstraints func(cmd cobraCmd)
+}
+
+type OkParentCmd struct {
+	Use                    string
+	Short                  string
+	Long                   string
+	Flags                  []OkFlag
+	Aliases                []string
+	EnforceFlagConstraints func(cmd cobraCmd)
+}
+
+func Init() cobraCmd {
+	return RegisterCommands(GetRootCommands())
+}
+
+// Go through commands recursively and build tree of commands
+func RegisterCommands(commands any) cobraCmd {
+	var rValue = reflect.Indirect(reflect.ValueOf(commands))
+
+	var parentCmd cobraCmd
+	var cmd cobraCmd
+	var okParentCmd *OkParentCmd
+	var okCmd *OkCmd
+
+	var subcommands = make([]*cobra.Command, 0)
+
+	for i := range rValue.NumField() {
```

Comment:
is reflection really needed here? It would be much simpler if you create a wrapper struct for a command and pass a slice of that to the register function.